compiler/hash-pipeline/src/lib.rs

```diff
@@ -213,6 +218,12 @@ where
             return (Err(err), compiler_state);
         }
 
+        // @@Temporary: if the mode is specified as `ast-gen` then we exit early to print the
+        //              generated AST tree
+        if matches!(self.settings.mode, CompilerMode::AstGen) {
+            return (result, compiler_state);
+        }
+
```

Comment:
It might be better if we have a more rigorous way of specifying which modes do what rather than knowing that in `ast-gen` mode, this will stop here. So it's a more programmatic way of specifying which modes do what rather than essentially *hard coding* what this mode does.

For example, A `check` mode could be:
```
check = ['ast-gen', 'typecheck', 'lint-x', ...]
```

So, then the pipeline machinery knows when to stop.